app/src/main/java/org/simple/clinic/instantsearch/InstantSearchEffectHandler.kt

```diff
@@ -73,6 +77,29 @@ class InstantSearchEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun checkIfPatientAlreadyHasAnExistingNHID(scheduler: Scheduler): ObservableTransformer<CheckIfPatientAlreadyHasAnExistingNHID, InstantSearchEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulers.computation())
+          .flatMap { patientRepository.patientProfile(it.patientId).subscribeOn(scheduler) }
+          .map { patientProfile ->
+            val patientUUID = patientProfile.get().patientUuid
+
+            if (hasNationalHealthID(patientProfile.get()))
+              PatientAlreadyHasAnExistingNHID(patientUUID)
+            else
+              PatientDoesNotHaveAnExistingNHID(patientUUID)
+          }
+    }
+  }
+
+  private fun hasNationalHealthID(patientProfile: PatientProfile): Boolean {
+    return patientProfile
+        .withoutDeletedBusinessIds()
```

Comment:
We could do this removal of delete `BusinessId` instances earlier in the Rx chain.

```kotlin
.map { it.get().withoutDeletedBusinessIds() }
```